### PR TITLE
Remove esql-core ParsingException

### PR DESCRIPTION
**Pull request by astefan**
Move fulltext related class packages to main `esql` project out of `esql-core`: querydsl, predicate and tests.

### CODE REVIEW

File: x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/planner/ExpressionTranslators.java [hunk 1]
```diff
@@ -71,18 +69,6 @@ private static Query translateField(RegexMatch e, String targetFieldName) {
         }
     }
 
-    public static class MultiMatches extends ExpressionTranslator<MultiMatchQueryPredicate> {
```
Comment (astefan) [inline x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/planner/ExpressionTranslators.java:74]: Moved to ExpressionTranslators in `esql`.

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/ParsingTests.java [hunk 2]
```diff
@@ -49,27 +49,27 @@ public class ParsingTests extends ESTestCase {
     );
 
     public void testCaseFunctionInvalidInputs() {
```
Comment (astefan) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/ParsingTests.java:51]: All these changes here started to be "visible" thanks to https://github.com/elastic/elasticsearch/pull/117153. And now we see the "danger" of running two "sister" classes in parallel in two different modules ;-).

**Review by costin — APPROVED** [review]
👍

**Comment by costin** [comment]
Makes sense to backport it.

**Pull request merged**